src/wallet/wallet.cpp

```diff
@@ -1417,6 +1418,57 @@ bool CWallet::IsHDEnabled() const
     return !hdChain.masterKeyID.IsNull();
 }
 
+void CWallet::EventuallyRescanAfterKeypoolTopUp() {
+    if (fSyncPausedUntilKeypoolExt) {
+
+        // for now, enable block requests and tip updates via the states
+        // this will only be sufficient as long as only a single wallet adjusts these stats
+        // otherwise the net-/validation-logic needs to poll the state over a signal (or similar)
```

Comment:
>Do you think this should be done in this PR or can we tackle this once we have multiple places where the sync may get paused?

Maybe, this PR is already pretty complicated, so I wouldn't want to tack this on in extra commits, but it could be nice if implemented to replace the existing "Add option to pause..." commits.